src/cargo.rs

```diff
@@ -432,6 +434,56 @@ pub fn rustdoc_flags(config: &Config) -> String {
     value
 }
 
+fn look_for_rustflags_in(path: &Path) -> Option<String> {
+    if let Ok(contents) = read_to_string(path) {
+        let value = contents.parse::<Value>().unwrap();
+
+        if let Some(build_value) = value.get("build") {
+            let build_table = build_value.as_table().unwrap();
+
+            if let Some(rustflags) = build_table.get("rustflags") {
+                let vec_of_flags: Vec<String> = rustflags
+                    .as_array()
+                    .unwrap()
+                    .into_iter()
+                    .map(|x| String::from(x.as_str().unwrap()))
+                    .collect();
+
+                return Some(vec_of_flags.join(" "));
+            }
+        }
+    }
+
+    None
+
+    // TODO handle [target.thumbv7em-none-eabihf] triplet flags etc.
+}
+
+fn build_config_path(base: &Path) -> PathBuf {
+    let mut config_path = PathBuf::new();
+    config_path.push(base);
+    config_path.push(".cargo");
+    config_path.push("config.toml");
+
+    config_path
+}
+
+fn gather_config_rust_flags(config: &Config) -> String {
+    if let Some(rustflags) = look_for_rustflags_in(&build_config_path(&config.root())) {
+        return rustflags;
+    }
+
+    if let Ok(cargo_home_config) = env::var("CARGO_HOME") {
+        if let Some(rustflags) =
+            look_for_rustflags_in(&build_config_path(&PathBuf::from(cargo_home_config)))
```

Comment:
If you make the change suggested with the `impl AsRef<Path>` above you can just pass in `cargo_home_config` without constructing the temporary `PathBuf`